torch/csrc/jit/script/ir_emitter.cpp

```diff
@@ -1871,11 +1884,48 @@ struct to_ir {
           {rhs},
           {},
           self);
-
+    } else if (lhsValue->type()->kind() == TypeKind::ClassType) {
+      // Call `__iadd__` so updates happen in place on class types
+      // https://docs.python.org/3/reference/datamodel.html#object.__iadd__
+      const char* magic_method_name;
+      const char* in_place_method_name;
+      const char* out_of_place_method_name;
+      std::tie(in_place_method_name, out_of_place_method_name) =
+          getAugMagicMethod(stmt);
+      const auto rhs = emitExpr(stmt.rhs());
+
+      if (lhsValue->type()->expect<ClassType>()->getMethod(
+              in_place_method_name)) {
+        magic_method_name = in_place_method_name;
+      } else {
+        magic_method_name = out_of_place_method_name;
+      }
+
+      // Insert call to magic method
+      auto sugared_magic_method = makeMagic(
+          magic_method_name,
+          std::make_shared<BuiltinFunction>(
+              getAugOp(stmt, lhsValue->type()), at::nullopt));
+
+      auto result = sugared_magic_method
+                        ->call(stmt.range(), method, {lhsValue, rhs}, {}, 0)
+                        ->asValue(stmt.range(), method);
+      // x += y is equivalent to x == x.__iadd__(y), so set the value to the
```

Comment:
this comment is no longer valid since at this point we could be doing either `__iadd__` or `__add__`